polkadot/runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -626,15 +651,26 @@ impl<T: Config> Pallet<T> {
 			ensure!(!votes_from_disabled_were_dropped, Error::<T>::BackedByDisabled);
 		}
 
+		// In `Enter` context (invoked during execution) we shouldn't have filtered any candidates
+		// due to a para having multiple cores assigned and no injected core index. They have been
+		// filtered during inherent data preparation (`ProvideInherent` context). Abort in such
+		// cases.
+		if context == ProcessInherentDataContext::Enter {
+			ensure!(
+				!dropped_elastic_scaling_candidates,
```

Comment:
Unrelated to this PR but why we check if we have dropped 'elastic scaling candidates' here, but we are fine dropping candidates without assigned core (which happens [here](https://github.com/paritytech/polkadot-sdk/blob/cb41758fe9a38ee26821bff93e615eb10848f7ad/polkadot/runtime/parachains/src/paras_inherent/mod.rs#L658))?

Reply:
Thanks for pointing this out!

prior to this PR, we just assumed that one para maps to one core. So we were only checking if the candidate's paraid has one scheduled core.
if it didn't, we'd return an error. if multiple candidates with the same paraid were supplied, we'd only check for one scheduled core.

The logic I added is more comprehensive and will work for proper elastic scaling also. But I decided to be more lenient and just filter out unscheduled candidates instead of erroring.

However, the check you mentioned should be I think extended to all dropped candidates 👍️ I don't see a reason why it should only be for some of them
because they should be all filtered by the block author. the other block importers should not do this

Reply:
done